src/UserSpaceInstrumentation/AllocateInTraceeTest.cpp

```diff
@@ -170,4 +170,57 @@ TEST(AllocateInTraceeTest, AutomaticAllocateAndFree) {
   waitpid(pid, nullptr, 0);
 }
 
+TEST(AllocateInTraceeTest, ReadSeccompModeOfThread) {
+  std::optional<int> seccomp_mode = ReadSeccompModeOfThread(getpid());
+  ASSERT_TRUE(seccomp_mode.has_value());
+  EXPECT_EQ(seccomp_mode.value(), SECCOMP_MODE_DISABLED);
+}
+
+TEST(AllocateInTraceeTest, SyscallInTraceeFailsBecauseOfSeccomp) {
+  pid_t pid = fork();
+  ORBIT_CHECK(pid != -1);
+  if (pid == 0) {
+    prctl(PR_SET_PDEATHSIG, SIGTERM);
+
+    // Transition to strict seccomp mode.
+    int64_t seccomp_result = syscall(SYS_seccomp, SECCOMP_SET_MODE_STRICT, 0, nullptr);
+    ORBIT_CHECK(seccomp_result == 0);
+
+    // Child just runs an endless loop.
+    volatile uint64_t counter = 0;
+    while (true) {
+      // Endless loops without side effects are UB and recent versions of clang optimize it away.
+      ++counter;
+    }
+  }
+
+  // Give the child enough time to execute the seccomp syscall.
+  std::this_thread::sleep_for(std::chrono::milliseconds{10});
+
+  int wstatus = 0;
+  pid_t waited_pid = waitpid(pid, &wstatus, WNOHANG);
+  ORBIT_CHECK(waited_pid != -1);
+  if (waited_pid != 0) {
+    // ORBIT_CHECK(seccomp_result == 0) failed in the child.
+    ORBIT_CHECK(WIFSIGNALED(wstatus) && WTERMSIG(wstatus) == SIGABRT);
+    ORBIT_ERROR("Child failed to call seccomp");
+    GTEST_SKIP();
+  }
```

Comment:
This test cannot be run on the CI&mdash;seccomp fails because the tests already run in seccomp mode 2. So I wanted to skip the tests in this case but apparently this doesn't work either. Any idea? Or should I just get rid of this test...?

Reply:
First for the sake of completeness: It is expected that seccomp runs in filter mode because that's one part of Docker's container security mechanism. In the past I already adjusted the list of allowed syscalls. It lives in `kokoro/builds/linux/seccomp.json`.

Back to the problem: The code looks good. I'm not sure what's wrong here. Currently I tend to disable seccomp security profiles for docker. Since the VM only hosts a single Docker container we won't get any security issues.

Would you be willing to try that? It can be disabled by changing line 335 in `kokoro/builds/build.sh` to `--security-opt seccomp=unconfined`.

Reply:
I'll try increasing the sleep first and then I'll try this.

Reply:
So disabling seccomp works. What should we do? Disable it even if it's just for this test?

Reply:
I think we should disable seccomp. It's probably worthwhile to have that test running. And there might be more coming along those lines.

Reply:
One more thing: If you decide to go with disabling seccomp, can you do me a favor and remove the seccomp.json file?

Reply:
Disabled seccomp and removed seccomp.json.